.github/workflows/release-tag.yml

```diff
@@ -0,0 +1,124 @@
+name: Tag release
+
+on:
+  push:
+    tags:
+      - 'v3.[0-9]+.[0-9]+'
+
+env:
+  NODE_ENV: production
+
+jobs:
+  documentation:
+    name: Documentation
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Check out documentation
+      uses: actions/checkout@v2
+      with:
+        ref: gh-pages
+        path: doc/public
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '~1.14'
+    - name: Download Go tool dependencies
+      run: |
+        cd tools
+        go mod download
+    - name: Make Mage
+      run: make tools/bin/mage
+    - name: Install JS SDK dependencies
+      run: tools/bin/mage jsSDK:deps
+    - name: Build JS SDK
+      run: tools/bin/mage jsSDK:clean jsSDK:build
+    - name: Install JS dependencies
+      run: tools/bin/mage js:deps
+    - name: File versioning
+      run: tools/bin/mage version:files
+    - name: Check for diff
+      run: tools/bin/mage git:diff
+    - name: Install documentation dependencies
+      run: tools/bin/mage docs:deps
+    - name: Build documentation
+      run: tools/bin/mage docs:build
+      env:
+        HUGO_BASE_URL: https://thethingsstack.io/
+    - name: Deploy documentation to Github pages
+      uses: JamesIves/github-pages-deploy-action@releases/v3
+      with:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        BRANCH: gh-pages
+        FOLDER: doc/public
+
+  release:
+    name: Release
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Login to Docker
+      uses: azure/docker-login@v1
+      with:
+        username: '${{ secrets.DOCKER_USERNAME  }}'
+        password: '${{ secrets.DOCKER_PASSWORD  }}'
+    - name: Install Snapcraft
+      run: sudo apt-get -yq --no-install-suggests --no-install-recommends install snapcraft
+    - name: Login to Snapcraft
+      env:
+        SNAPCRAFT_LOGIN: ${{ secrets.SNAPCRAFT_LOGIN }}
+      run: snapcraft login --with <(printf "$SNAPCRAFT_LOGIN")
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '~14'
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '~1.14'
+    - name: Download Go dependencies
+      run: go mod download
+    - name: Download Go tool dependencies
+      run: |
+        cd tools
+        go mod download
+    - name: Make Mage
+      run: make tools/bin/mage
+    - name: Install JS SDK dependencies
+      run: tools/bin/mage jsSDK:deps
+    - name: Build JS SDK
+      run: tools/bin/mage jsSDK:clean jsSDK:build
+    - name: Install JS dependencies
+      run: tools/bin/mage js:deps
+    - name: Build frontend
+      run: tools/bin/mage js:clean js:build
+    - name: File versioning
+      run: tools/bin/mage version:files
+    - name: Check for diff
+      run: tools/bin/mage git:diff
+    - name: Determine date
```

Comment:
We should also build the offline docs here

Reply:
Can you clarify? It's not something present in `.travis.yml`, right?

Reply:
It's `$MAGE docs:build`